### PR TITLE
Added GitHub Login

### PR DESCRIPTION
**Pull request by roeeyn**
Added GitHub Login

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -9,7 +9,9 @@
     "@testing-library/user-event": "^7.2.1",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
-    "react-scripts": "3.4.1"
+    "react-router-dom": "^5.2.0",
+    "react-scripts": "3.4.1",
+    "styled-components": "^5.1.1"
```
Comment (amy-lei) [inline client/package.json:14]: This is my first time using styled-components, but will it limiting? I guess my main concern is just variables. Im currently using sass for the card components, and Ive added a lot of variables for consistent coloring/sizing.

File: client/src/components/GitHubHome.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  const handleLogout = () => {
+    dispatch({
+      type: "LOGOUT"
+    });
+  } 
+
+  return (
+    <Wrapper>
+      <div className="container">
+        <button onClick={()=> handleLogout()}>Logout</button>
```
Comment (amy-lei) [inline client/src/components/GitHubHome.js]: i think since we're not passing any args, it can just be `onClick={handleLogout}`

**Review by amy-lei — CHANGES_REQUESTED** [review]
This is great! I left a comment about using `styled-components` in general, and one for a minor change.

I think moving forward, we also would populate the DB with the user if this is their first time logging in, otherwise, get the user data from the DB and return it as well. Maybe that can be a separate task and a separate PR since this is good work!

**Review by amy-lei — APPROVED** [review]

**Comment by amy-lei** [comment]
LGTM now! As i mentioned above, we'll need to create a User instance and fetch user data & posts, but i think that can be for another PR :)

**Comment by yahengsu** [comment]
looks good to me, but can we move the `.css` files out to a `styles/` folder instead of keeping them in `components/`?

**Comment by rch-liu** [comment]
LGTM 👍

**Pull request merged**